lib/bamboo/sent_email.ex

```diff
@@ -89,7 +89,7 @@ defmodule Bamboo.SentEmail do
 
   defp do_get(id) do
     Enum.find all, nil, fn(email) ->
-      get_id(email) == id
+      get_id(email) == String.downcase(id)
```

Comment:
While that's true for a lowercase rewrite, it’s not case-insensitive without this line. It would fail for an uppercase rewrite.